apps/dashboard/src/components/contract-components/hooks.ts

```diff
@@ -300,7 +300,10 @@ async function fetchPublisherProfile(publisherAddress?: string | null) {
     getDashboardChainRpc(polygon.id, undefined),
   );
   invariant(publisherAddress, "address is not defined");
-  return await sdk.getPublisher().getPublisherProfile(publisherAddress);
+  return (await sdk
+    .getPublisher()
+    // todo: remove type-casting once we have replaced this method
+    .getPublisherProfile(publisherAddress)) as ProfileMetadata;
```

Comment:
would love to do that at the same time as this in the PR? what would that take? if it's too much we can ship this first but the real win is removing this

Reply:
I will do it in this PR. there "no rush" getting this one out anyway

Reply:
hmm I think we still have to type-cast one way or another? Because we are getting the profile metadata from an IPFS uri  (by calling the method `getPublisherProfileUri` on our Publisher contract)

Maybe I should do a separate PR where I replace all the v4 publisher code inside `src/components/contract-components/hooks.ts`

wdyt @jnsdls

Reply:
@kien-ngo if you want to do that, please go ahead - otherwise I'll do it today and we can merge this first